eng/common/scripts/Generate-DocIndex.ps1

```diff
@@ -0,0 +1,147 @@
+    "python"     = @("Python", "python")
+    "c"          = @("C", "c")
+    "cpp"        = @("CPP", "cpp")
+}
+
+# The sequence of Bom bytes differs by different encoding. 
+# The helper function here is only to strip the utf-8 encoding system as it is used by blob storage list api.
+# Return the original string if not in BOM utf-8 sequence.
+function removeBomFromString($bomAwareString) {
+    if ($bomAwareString.length -le 3) {
+        return $bomAwareString
+    }
+    $bomPatternByteArray = [byte[]] (0xef, 0xbb, 0xbf)
+    # The default encoding for powershell is ISO-8859-1, so converting bytes with the encoding.
+    $bomAwareBytes = [Text.Encoding]::GetEncoding(28591).GetBytes($bomAwareString.Substring(0, 3))
+    if (@(Compare-Object $bomPatternByteArray $bomAwareBytes -SyncWindow 0).Length -eq 0) {
+        return $bomAwareString.Substring(3)
+    }
+    return $bomAwareString
+}
+Write-Verbose "Name Reccuring paths with variable names"
+$DocFxTool = "${RepoRoot}/docfx/docfx.exe"
+$DocOutDir = "${RepoRoot}/docfx_project"
+
+Write-Verbose "Initializing Default DocFx Site..."
+& "${DocFxTool}" init -q -o "${DocOutDir}"
+
+Write-Verbose "Copying template and configuration..."
+New-Item -Path "${DocOutDir}" -Name "templates" -ItemType "directory" -Force
+Copy-Item "${DocGenDir}/templates/*" -Destination "${DocOutDir}/templates" -Force -Recurse
+Copy-Item "${DocGenDir}/docfx.json" -Destination "${DocOutDir}/" -Force
+$YmlPath = "${DocOutDir}/api"
+New-Item -Path $YmlPath -Name "toc.yml" -Force
+
+Write-Verbose "Reading artifact from storage blob ..."
+$metadata = GetMetaData -lang $lang
+$langRegex = $LangMapping[$lang][1]
+$titleRegex = $LangMapping[$lang][0]
+$blobDirectoryRegex = "^$langRegex/(.*)/$"
+$pageToken = ""
+# Used for sorting the toc display order
+$orderSet = @{}
+$defaultGroupId = "com.azure"
+# This is a pagnation call as storage only return 5000 results as maximum.
+Do {
+    $resp = ""
+    if (!$pageToken) {
+        # First page call.
+        $resp = Invoke-RestMethod -Method Get -Uri "https://azuresdkdocs.blob.core.windows.net/%24web?restype=container&comp=list&prefix=$langRegex%2F&delimiter=%2F"
+    }
+    else {
+        # Next page call
+        $resp = Invoke-RestMethod -Method Get -Uri "https://azuresdkdocs.blob.core.windows.net/%24web?restype=container&comp=list&prefix=$langRegex%2F&marker=$pageToken"
+    }
+    # Storage returns some weired encoded string at the begining of response which needs to cutoff before use.
+    
+    # Convert to xml documents.
+    $xmlDoc = [xml](removeBomFromString $resp)
+    foreach ($elem in $xmlDoc.EnumerationResults.Blobs.BlobPrefix) {
+        # What service return like "dotnet/Azure.AI.Anomalydetector/", needs to fetch out "Azure.AI.Anomalydetector"
+        $artifact = $elem.Name -replace $blobDirectoryRegex, '$1'
+        if ($blobPath -and ($blobPath -ne $artifact)) {
+            # $blobRentionArray = $blobPath -split "," | ForEach-Object { $_.Trim() }   
```

Comment:
Looks like this is trying to hack around the c case. Lets talk about that case a little more I have some potential ideas.